inbox/mailsync/backends/imap/monitor.py

```diff
@@ -103,8 +102,9 @@ def save_folder_names(self, db_session, raw_folders):
 
         # Create new folders
         for raw_folder in raw_folders:
-            Folder.find_or_create(db_session, account, raw_folder.display_name,
-                                  raw_folder.role)
+            folder = Folder.find_or_create(db_session, account, raw_folder.display_name,
+                                           raw_folder.role)
+            Folder.update_role(folder, raw_folder.role)
```

Comment:
Hmm - does this need to update the Category associated with the Folder as well? Updating only the folder might not have an impact on the API response to /folders or /labels.